jdbc-yugabytedb/src/main/java/org/postgresql/Driver.java

```diff
@@ -450,12 +451,89 @@ public Connection getResult(long timeout) throws SQLException {
    * thread without enforcing a timeout, regardless of any timeout specified in the properties.
    *
    * @param url the original URL
-   * @param props the parsed/defaulted connection properties
+   * @param properties the parsed/defaulted connection properties
    * @return a new connection
    * @throws SQLException if the connection could not be made
    */
-  private static Connection makeConnection(String url, Properties props) throws SQLException {
-    return new PgConnection(hostSpecs(props), user(props), database(props), props, url);
+  private static Connection makeConnection(String url, Properties properties) throws SQLException {
+    LoadBalanceProperties lbprops = new LoadBalanceProperties(url, properties);
+    if (lbprops.hasLoadBalance()) {
+      Connection conn = getConnectionBalanced(lbprops);
+      if (conn != null) {
+        return conn;
+      } else {
+        LOGGER.log(Level.WARNING, "Failed to apply load balance. Trying normal connection");
+        // but remember to use stripped url and props
+        url = lbprops.getStrippedURL();
+        properties = lbprops.getStrippedProperties();
+      }
+    }
+    return new PgConnection(hostSpecs(properties), user(properties), database(properties), properties, url);
+  }
+
+  private static Connection getConnectionBalanced(LoadBalanceProperties lbprops) throws SQLException {
```

Comment:
getConnectionBalanced -> getBalancedConnection

Reply:
No, it is deliberately named like that. There is a getConnection method. I have introduced a 'balanced' variant of that. If you keep it 'getBalancedConnection' it will sound more like an attribute of the connection which is not the case here.